packages/console/src/contexts/AppDataProvider.tsx

```diff
@@ -35,18 +36,19 @@ function AppDataProvider({ children }: Props) {
 
   useEffect(() => {
     const getEndpoint = async () => {
-      if (!currentTenantId) {
-        return;
-      }
-
+      setIsLoading(true);
       const { user } = await ky
         .get(new URL(`api/.well-known/endpoints/${currentTenantId}`, adminTenantEndpoint))
         .json<{ user: string }>();
       setUserEndpoint(new URL(user));
     };
 
+    if (!currentTenantId || isLoading || userEndpoint) {
```

Comment:
nit: Why `userEndpoint`? I don't quite get this one

Reply:
yeah you asked this before. i think this is quite confusing - i'll open another PR to refactor this LOG-6604